app/src/main/java/com/heapixLearn/discovery/Server/Contacts/ServerContactManager.java

```diff
@@ -0,0 +1,96 @@
+package com.heapixLearn.discovery.Server.Contacts;
+
+
+import com.heapixLearn.discovery.DB.AuthStore;
+import com.heapixLearn.discovery.RunnableWithObject;
+import com.heapixLearn.discovery.Server.Controller;
+import com.heapixLearn.discovery.Server.ServerAnswer;
+import com.heapixLearn.discovery.Server.TypeOfServerError;
+import com.heapixLearn.discovery.Server.UserApi;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class ServerContactManager {
+
+    private UserApi userApi;
+    private AuthStore authStore;
+
+    public ServerContactManager(){
+        userApi = Controller.getApi();
+        authStore = AuthStore.getInstance();
+    }
+
+    public void readContacts(final RunnableWithObject<List<ServerContact>> onSuccess, final RunnableWithObject<TypeOfServerError> onFailure){
+        userApi.readContacts(authStore.getToken()).enqueue(new Callback<List<ServerContact>>() {
+            @Override
+            public void onResponse(Call<List<ServerContact>> call, Response<List<ServerContact>> response) {
+                if (isExist(response.body())){
+                    onSuccess.init(response.body());
+                } else {
+                    onFailure.init(TypeOfServerError.SERVER_ERROR);
+                    onFailure.run();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<List<ServerContact>> call, Throwable t) {
+                onFailure.init(TypeOfServerError.SERVER_ERROR);
+                onFailure.run();
+            }
```

Comment:
@Agent00krosh where is error of wrong credentials? How should the app know that it should to logout?